great_expectations/checkpoint/v1_checkpoint.py

```diff
@@ -120,8 +129,97 @@ def run(
         batch_parameters: Dict[str, Any] | None = None,
         expectation_parameters: Dict[str, Any] | None = None,
     ) -> CheckpointResult:
-        raise NotImplementedError
+        run_id = RunIdentifier(run_time=dt.datetime.now(dt.timezone.utc))
+        run_results = self._run_validation_definitions(
+            batch_parameters=batch_parameters,
+            expectation_parameters=expectation_parameters,
+            run_id=run_id,
+        )
+        self._run_actions(run_results=run_results)
+
+        return CheckpointResult(
+            run_id=run_id,
+            run_results=run_results,
+            checkpoint_config=self,
+            validation_result_url=None,  # TODO: Need to plumb from actions
+        )
+
+    def _run_validation_definitions(
+        self,
+        batch_parameters: Dict[str, Any] | None,
+        expectation_parameters: Dict[str, Any] | None,
+        run_id: RunIdentifier,
+    ) -> Dict[ValidationResultIdentifier, ExpectationSuiteValidationResult]:
+        run_results: Dict[ValidationResultIdentifier, ExpectationSuiteValidationResult] = {}
+        for validation_definition in self.validation_definitions:
+            key = self._build_result_key(validation_definition=validation_definition, run_id=run_id)
+            validation_result = validation_definition.run(
+                batch_definition_options=batch_parameters,
+                evaluation_parameters=expectation_parameters,
+            )
+            run_results[key] = validation_result
+
+        return run_results
+
+    def _build_result_key(
+        self, validation_definition: ValidationConfig, run_id: RunIdentifier
+    ) -> ValidationResultIdentifier:
+        return ValidationResultIdentifier(
+            expectation_suite_identifier=ExpectationSuiteIdentifier(
+                name=validation_definition.suite.name
+            ),
+            run_id=run_id,
+            batch_identifier=validation_definition.active_batch_id,
+        )
+
+    def _run_actions(
+        self, run_results: Dict[ValidationResultIdentifier, ExpectationSuiteValidationResult]
+    ) -> None:
+        # TODO: Open to implement once actions are updated to run on aggregated results
+        pass
 
     @public_api
     def save(self) -> None:
         raise NotImplementedError
+
+
+class CheckpointResult(BaseModel):
+    run_id: RunIdentifier
+    run_results: Dict[ValidationResultIdentifier, ExpectationSuiteValidationResult]
+    checkpoint_config: Checkpoint
+    validation_result_url: Optional[str] = None
+    success: Optional[bool] = None
```

Comment:
Should always be set by the root validator rather than a user (but so should this entire object)